### ISSUE TITLE
Allow me to search for markdown headers without manually inserting `-` delimiters

### ISSUE THREAD

**Issue by joyceerhl** [microsoft/vscode#174999]
Testing #174884

Currently if I have a header like so 

```md
# A header with spaces in the title
```

I have to search for it by typing the following when triggering header autocomplete

```
#a-header-with-spaces-in-the-title
```

I'd like to be able to type

```
#a header with spaces in the title
```

and have the completion still show up.

**Comment by mjbvz** [comment]
I agree but am not yet sure if this belongs in the suggest widget or a new UI. Using the suggest widget is a little weird because the inserted text has to be slugified

I'm potentially imagining a `Insert header link` command that shows a quick pick with all headers for fuzzy matching. Selecting one would insert the full link, similar to how `Markdown: Insert link to file in workspace` works
